File: 03.Functions-And-Statements-Exercise/palindromeIntegers.js
function checkPalindrome(arrayOfNumbers) {

    function isPalindrome(number) {
        let numberAsString = String(number)
        let numberAsArray = numberAsString.split("")
        let reversedArray = numberAsArray.reverse()
        let reversedNumberAsString = reversedArray.join("")
        let reversedNumber = Number(reversedNumberAsString)
        return number === reversedNumber
    }

    for (const num of arrayOfNumbers) {
        if (isPalindrome(num)) {
            console.log('true')
        } else {
            console.log('false')
        }
    }
}

checkPalindrome([123,323,421,121])
checkPalindrome([32,2,232,1010])

File: 03.Functions-And-Statements-Exercise/palindrome_integers.py
def check_palindrome(list_of_numbers):

    def is_palindrome(number):
        number_as_string = str(number)
        reversed_string = number_as_string[::-1]
        return int(reversed_string) == number
    
    for num in list_of_numbers:
        if is_palindrome(num):
            print('true')
        else:
            print('false')

check_palindrome([123,323,421,121])
check_palindrome([32,2,232,1010])
